lib/Arch/X86/Semantics/GET_PC.cpp

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2022 Trail of Bits, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+template <typename D>
+DEF_SEM(GET_PC, D dst) {
+  addr_t pc = Read(REG_PC);
+  Write(dst, Read(ReadPtr<D>(pc)));
+  return memory;
+}
+
+DEF_ISEL(GET_PC_16) = GET_PC<M16W>;
```

Comment:
I don't really know what these type names like `M16W` and `RnW` refer to. I just copied this from the POP implementation.

Reply:
Oh, there's some information in the "life of an instruction" doc. I'll check that out.

Reply:
[07abb69](https://github.com/lifting-bits/remill/pull/610/commits/07abb69d3bc0daa72c4dcaf277b567c1aa4b7f45)